_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,411 @@
+- Use a specific storage class? (leave empty)
+
+**NOTE**: You can leave the Docker repository name blank for running Kubernetes locally, but a repository will be required for the cloud deployment, so go ahead and create a [Docker Hub](https://hub.docker.com/) personal account, and the image pull configuration will be ready for both local and cloud deployments.
+
+Build the `gateway`, `store` and `blog` services container images with Jib. For example, for the `gateway` service:
+
+```bash
+cd ../gateway
+./gradlew bootJar -Pprod jib -Djib.to.image=<docker-repo-name>/gateway
+```
+
+Check the images were uploaded to [DockerHub](https://hub.docker.com), and navigate to the project root folder in the terminal for the next step.
+
+## Configure authentication with OpenID Connect and Okta
+
+One more configuration step before running the architecture locally, let's configure Okta for authentication.
+
+{% include setup/cli.md type="jhipster" %}
+
+The settings from the generated `.okta.env` must be added to the `k8s/registry-k8s/application-configmap.yml`:
+
+```yml
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+```
+
+Enable the OIDC authentication in the `jhipster-registry` service by adding the `oauth2` profile in the `k8s/registry-k8s/jhipster-registry.yml` file:
+
+```yml
+- name: SPRING_PROFILES_ACTIVE
+  value: prod,k8s,oauth2
+```
+
+## Run locally with Minikube
+
+Install [Minikube](https://minikube.sigs.k8s.io/docs/start/) and [`kubectl`](https://kubernetes.io/docs/tasks/tools/).
+
+For Minkube, you will need at least 2 CPUs. Start MiniKube with your number of CPUs:
+
+```bash
+cd k8s
+minikube --cpus <ncpu> start
+```
+Minikube will log the Kubernetes and Docker versions on start:
+
+```
+Preparing Kubernetes v1.23.3 on Docker 20.10.12 ...
+```
+
+For the `store-mongodb` deployment to work, the property `Service.spec.publishNotReadyAddresses` was required, instead of the annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints`, as the latter was deprecated in [Kubernetes release 1.11](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.11.md#kubernetes-111-release-notes).
+
+In the k8s folder, edit `store-mongodb.yml` and add the `publishNotReadyAddresses: true` property to the `spec`:
+
+```yml
+# Headless service for DNS record
+apiVersion: v1
+kind: Service
+metadata:
+  name: store-mongodb
+  namespace: demo
+spec:
+  type: ClusterIP
+  clusterIP: None
+  publishNotReadyAddresses: true
+  ports:
+    - name: peer
+      port: 27017
+  selector:
+    app: store-mongodb
+```
+
+Then deploy the application to Minikube, in the `k8s` directory, run:
+
+```bash
+./kubectl-apply.sh -f
```

Comment:
On my Intel Mac, the PostgreSQL database fails to startup.

<img width="1442" alt="Screen Shot 2022-05-13 at 17 57 23" src="https://user-images.githubusercontent.com/17892/168402598-cb556d0c-4f8e-45b3-92c0-e36e382c0bf5.png">

This makes it so the gateway can't startup.

<img width="1442" alt="Screen Shot 2022-05-13 at 17 58 33" src="https://user-images.githubusercontent.com/17892/168402617-f17ddb96-8db4-448e-b302-435f554982be.png">

If you think it might help, I can update my original "Reactive Java Microservices" blog to use the latest JHipster.

Reply:
I will qa with JHipster 7.8.1 and see what issues come up.

Reply:
I generated the app with the repo jdl, Jhipster 7.8.1, and generated the k8s, updated config for okta oauth2 and set the store-mongodb publishNotReadyAddresses, and minikube worked.
Before a success run, I had network issues and minikube was unresponsive, minikube logs showed "error getting server version."

![jhipster-8](https://user-images.githubusercontent.com/2080632/168902598-a753fa73-d1b7-459f-8211-0ed599a3cb13.png)

Reply:
OK, I'll just continue with the Digital Ocean part.